### PR TITLE
islandora_webform issue #23

### PR DESCRIPTION
**Pull request by patdunlavey**
See issue #23. There are a lot of threads that are hard to follow. I want to get this merged so we can all be working from the same code.

Basically, this extends on this pull request: https://github.com/commonmedia/islandora_webform/pull/26, fixing a couple minor bugs and providing api documentation of hook_iwi_set_form_alter().

### CODE REVIEW

**Comment by patdunlavey** [comment]
@DiegoPino any thoughts on merging this pull request?

**Comment by DiegoPino** [comment]
@patdunlavey, just for clarity, why not make individual pulls? It's difficult to see how all this interact, and also by being a merge from your master, all the history gets lost or not?

File: submodules/islandora_webform_ingest/includes/utilities.inc [hunk 1]
```diff
@@ -507,8 +507,13 @@ function iwi_set_form_element_byhash(&$form, &$form_state, $elementspath, $input
                 if (!isset($special_element_children[$offset])) {
                   // If not already cloned for this offset clone it.
                   $new_tab = clone $tab;
-                  $set_element_properties = function ($element) {
+                  $set_element_properties = function ($element) use ($special_xmlform_elements) {
                     $element->default_value = NULL;
+                    foreach($element->children as $key => &$child) {
+                      if($key !== 0 && in_array($child->controls['#type'], $special_xmlform_elements) ) {
```
Comment (patdunlavey) [inline submodules/islandora_webform_ingest/includes/utilities.inc]: This is meant to address https://github.com/commonmedia/islandora_webform/issues/27. We only are trying to clone the structure of the descendent special xmlform elements, not their data. So there is no reason to keep more than one.

**Comment by patdunlavey** [comment]
@DiegoPino that would be better, but I'm not sure how to go back and tease this apart into independent pull requests that could stand on their own.

**Comment by patdunlavey** [comment]
Accepting this pull request will:
resolve #14
resolve #27

**Comment by DiegoPino** [comment]
@patdunlavey, fine for me. Thanks
